src/routes/delegates/pipes/create-delegate.dto.validation.pipe.ts

```diff
@@ -0,0 +1,31 @@
+import { HttpStatus, Injectable, PipeTransform } from '@nestjs/common';
+import { ValidateFunction } from 'ajv';
+import { GenericValidator } from '../../../domain/schema/generic.validator';
+import { JsonSchemaService } from '../../../domain/schema/json-schema.service';
+import { CreateDelegateDto } from '../entities/create-delegate.dto.entity';
+import { createDelegateDtoSchema } from '../entities/schemas/create-delegate.dto.schema';
+
+@Injectable()
+export class CreateDelegateDtoValidationPipe
+  implements PipeTransform<any, CreateDelegateDto>
+{
+  private readonly isValid: ValidateFunction<CreateDelegateDto>;
+
+  constructor(
+    private readonly genericValidator: GenericValidator,
+    private readonly jsonSchemaService: JsonSchemaService,
+  ) {
+    this.isValid = this.jsonSchemaService.getSchema(
+      'https://safe-client.safe.global/schemas/delegates/create-delegate.dto.json',
+      createDelegateDtoSchema,
+    );
+  }
+  transform(data: any): CreateDelegateDto {
+    try {
+      return this.genericValidator.validate(this.isValid, data);
+    } catch (err) {
+      err.status = HttpStatus.BAD_REQUEST;
+      throw err;
+    }
+  }
```

Comment:
Have you considered something like the `DataSourceErrorFilter`?

Basically we can work only with exceptions at this level and ignore the HTTP body/status. The Error Filter would then, depending on the exception caught assign the correct response payload.

Reply:
Yes, I agree in general terms, but do you think this could be done in a separate PR? I think it could be useful to define a `ValidationErrorFilter` similar to `DataSourceErrorFilter` and attach it to the application, but I also think we should differentiate client validation errors from errors when validating provider responses (Transaction Service, Config Service, exchanges, etc.). 

If that's the case we would need to modify a lot of tests as well, and add a couple of classes, so maybe a new PR fits better, what do you think?

Reply:
> Yes, I agree in general terms, but do you think this could be done in a separate PR? I think it could be useful to define a `ValidationErrorFilter` similar to `DataSourceErrorFilter` and attach it to the application, but I also think we should differentiate client validation errors from errors when validating provider responses (Transaction Service, Config Service, exchanges, etc.).
> 
> If that's the case we would need to modify a lot of tests as well, and add a couple of classes, so maybe a new PR fits better, what do you think?

Can be done in a different PR. And doing something like the `DataSourceErrorFilter` doesn't mean that you need to output the same error as with the current `DataSourceErrorFilter`. I was only talking about the mechanism surrounding catching types of errors to return a specific response.